### DISCUSSION TITLE
Feature: 管理者以外によるグループの設定

### DISCUSSION THREAD

**Discussion by shield-9** [growilabs/growi#7709] [Ideas / Requests] [answered]
現状のグループ機能は、作成・変更のたび管理者による作業が必要な仕様です。
そのため、数百ユーザのいる環境では管理者の負担が大きすぎるという判断の元、グループ機能の運用を断念せざるを得ない状況です。

そのために、グループの設定画面だけを一般利用者が使えるようにする機能が欲しいです。

**Comment by yuki-takei** [comment]
@shield-9 ご提案ありがとうございます。ユーザー数が多い環境でどのような課題が出るのかは非常に興味深いです。GROWIをより良い物にしていくためにもどんどんフィードバックいただければ幸いです。

パッと思いつく設計だと「アビリティ」という概念を追加して、「ユーザーグループの設定可」「カスタマイズの設定可」のようなアビリティを複数個ユーザーまたはグループに付けられるといいのかなという所感です。他にこれがよいのではという案がありましたら返信ください。(他の方からの意見も募集します)


## 蛇足
別の側面からの策ですが、 #491 のような機能の要望が昔出て、先月あたりから設計は開始したところです。

[dev wiki: /Web会議室/ExternalGroup](https://dev.growi.org/5cd3b02d091bdc00486b2f42)

こちらよりロール実装の方がマンパワーがあればどうにかできる幅が広がるので汎用性ありますかねえ。

**Comment by shield-9** [comment]
こちらこそ、これから利用者数が1000人を超えそうなところで、より様々なケースが出てくると思いますので、積極的にフィードバック(&内製パッチについてはコントリビューションも)させていただきます 🙇 

アビリティはグループに付与できれば、管理権限のためには十分かなと考えています。理由は、1ユーザーのみが所属するグループを作成することで、ユーザーへの付与と同等のことが実現できるからです。

一方、思い付きですが、ユーザーとグループの両方に付与できる汎用的な実装であれば、この仕組みを使ってページの閲覧・編集権限の管理ができるかなと。2つ以上のグループやユーザーに対してページの権限を付与したいというニーズも確実にあると思うので、それも同一の仕組み上に実装できたらいいなと思いました。

**Comment by yuki-takei** [comment, marked as answer]
4年越しの要望になりますが、v6.3.2 で LDAP/Keycloak からのグループ同期が実装されました。
https://docs.growi.org/ja/admin-guide/upgrading/63x.html
